control-plane/stor-port/src/types/v0/store/snapshots/volume.rs

```diff
@@ -0,0 +1,188 @@
+use super::{replica::ReplicaSnapshot, SnapshotId, SnapshotSpec};
+use crate::types::v0::{
+    store::{AsOperationSequencer, OperationSequence, SpecStatus, SpecTransaction},
+    transport::VolumeId,
+};
+use chrono::{DateTime, Utc};
+use pstor::{ApiVersion, ObjectKey, StorableObject, StorableObjectType};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// User specification of a volume snapshot.
+/// todo: is e-derivations a better way of doing this?
+pub type VolumeSnapshotUserSpec = SnapshotSpec<VolumeId>;
+
+/// State of the VolumeSnapshotSpec Spec.
+pub type VolumeSnapshotSpecStatus = SpecStatus<()>;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
+pub struct VolumeSnapshot {
+    /// Status of the volume snapshot.
+    status: VolumeSnapshotSpecStatus,
+    /// User specification of the snapshot.
+    spec: VolumeSnapshotUserSpec,
+    /// Control-plane related information of the snapshot (book-keeping).
+    metadata: VolumeSnapshotMeta,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
+pub struct VolumeSnapshotMeta {
+    #[serde(skip)]
+    sequencer: OperationSequence,
+    /// Record of the operation in progress.
+    operation: Option<VolumeSnapshotOperationState>,
+
+    /// Creation timestamp of the snapshot (set after creation time).
+    creation_timestamp: Option<DateTime<Utc>>,
+    /// Size of the snapshot (typically follows source size).
+    size: u64,
+    /// Transaction Id that defines this snapshot when it is created.
+    txn_id: String,
+    /// Replicas which "reference" to this snapshot as its parent, indexed by the transaction
+    /// id when they were attempted.
+    /// The "actual" snapshots can be accessed by the key `txn_id`.
```

Comment:
Should the key be `txn_id` ? In that case what does the above `txn_id` field represent, the last transaction id?

Reply:
Cool. Thanks.